### PR TITLE
Add headers and recursive depth to file_list

### PR DESCRIPTION
**Pull request by micahjmartin**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Enhance the file_list tome with the following features
- Added support for recursive depth in file listing
- Added permission check for listing directories
- Add headers to top of printout
- Print full file path

### CODE REVIEW

**Review by hulto — CHANGES_REQUESTED** [review]
Please move recursion to the file_tree tome.
We'll work on adding rwx to eldritch natively but looks good to me.

File: tavern/tomes/file_list/main.eldritch [hunk 1]
```diff
@@ -1,17 +1,111 @@
...
+    # If the user isnt root and the group doesnt own the file, clear the group byte    
+    if not root and f["group"] not in (str(usernfo["egid"]), str(usernfo["gid"])):
+        f_group = 0
+
+    if (f_group & PERM_READ) | (f_user & PERM_READ):
+        return True
+    return False
+
+def glob(s, pattern):
+    #if pattern == s or pattern == "*":
```
Comment (hulto) [inline tavern/tomes/file_list/main.eldritch:43]: Please remove dead code.

**Review by hulto — APPROVED** [review]
One nit but otherwise LGTM

File: tavern/tomes/file_list/main.eldritch [hunk 2]
```diff
@@ -1,17 +1,115 @@
...
+
+SEP = "/"
+if sys.get_os().get("platform", "") == "PLATFORM_WINDOWS":
+    SEP = "\\"
+
+def can_read(f):
+    """Return true if the user can read this dir/file
+    """
+
+    # Until we get permissions on windows, just go ahead and try to read
```
Comment (micahjmartin) [inline tavern/tomes/file_list/main.eldritch:17]: @hulto What do you think about this? better to try and fail then not try, right?

Comment (hulto) [reply tavern/tomes/file_list/main.eldritch:17]: Yeah for now just send it 🤷

File: tavern/tomes/file_list/metadata.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: List files
-description: List the files and directories found at the path
+description: List the files and directories found at the path. Supports basic glob functionality. Does not glob more than one level
 author: hulto
 support_model: FIRST_PARTY
 tactic: RECON
```
Comment (hulto) [inline tavern/tomes/file_list/metadata.yml:5]: Could you add an example of globbing to the placeholder.
Just `/etc/open*` probably.

**Review by hulto — CHANGES_REQUESTED** [review]
One docs ask.

**Review by hulto — APPROVED** [review]
APPROVED!

**Pull request merged**